lib/execution_engine2/utils/arg_processing.py

```diff
@@ -33,4 +38,124 @@ def parse_bool(putative_bool: Union[str, bool, int, float, None]) -> bool:
+    :returns: the item.
+    """
+    if not item:
+        raise ValueError(f"{item_name} cannot be a value that evaluates to false")
+    return item
+
+
+def not_falsy_in_iterable(
+    iterable: Optional[Iterable[T]], name: str, allow_none: bool = False
+) -> Optional[Iterable[T]]:
+    """
+    Check that an iterable is not None and contains no falsy items. Empty iterables are accepted.
+    :param iterable: the iterable to check.
+    :param name: the name of the iterable to be used in error messages.
+    :param allow_none: allow the iterable to be None - in this case return None. The contents of
+        the iterable may not be None.
+    :returns: the iterable.
+    :raises ValueError: if the iterable is None or contains falsy items.
+    """
+    # probably need to allow for 0 as an option
+    if iterable is None:
+        if allow_none:
+            return None
+        raise ValueError(f"{name} cannot be None")
+    for i, item in enumerate(iterable):
+        not_falsy(item, f"Index {i} of iterable {name}")
+    return iterable
+
+
+def _contains_control_characters(string: str) -> bool:
+    """
+    Check if a string contains control characters, as denoted by the Unicode character category
+    starting with a C.
+    :param string: the string to check.
+    :returns: True if the string contains control characters, False otherwise.
+    """
+    # make public if needed
+    # See https://stackoverflow.com/questions/4324790/removing-control-characters-from-a-string-in-python  # noqa: E501
+    for c in string:
+        if unicodedata.category(c)[0] == "C":
+            return True
+    return False
+
+
+def _no_control_characters(string: str, name: str) -> str:
+    """
+    Checks that a string contains no control characters and throws an exception if it does.
+    See :meth:`contains_control_characters` for more information.
+    :param string: The string to check.
+    :param name: the name of the string to include in any exception.
+    :raises IncorrectParamsException: if the string contains control characters.
+    :returns: the string.
+    """
+    # make public if needed
+    if _contains_control_characters(string):
+        raise IncorrectParamsException(name + " contains control characters")
+    return string
+
+
+def check_string(
+    string: Optional[str], name: str, max_len: int = None, optional: bool = False
+) -> Optional[str]:
+    """
+    Check that a string meets a set of criteria:
+    - it is not None or whitespace only (unless the optional parameter is specified)
+    - it contains no control characters
+    - (optional) it is less than some specified maximum length
+    :param string: the string to test.
+    :param name: the name of the string to be used in error messages.
+    :param max_len: the maximum length of the string.
+    :param optional: True if no error should be thrown if the string is None.
+    :returns: the stripped string or None if the string was optional and None or whitespace only.
```

Comment:
I see the string returns the stripped version, but should it? I would almost expect that to either be an option, or just to return the same string I sent to it.

Reply:
After a whooole lot of experience using these functions it absolutely should. I've used functions like this hundreds of times in code since 2016 or so and I always want the strip. The use case for most of these functions is checking user inputs for functions or constructors where whitespace around strings is to be ignored.